Assets/Xbox Live/Scripts/Leaderboard.cs

```diff
@@ -503,18 +504,14 @@ private void OnPlayerSignOut(XboxLiveUser xboxLiveUser, XboxLiveAuthStatus authS
                 this.xboxLiveUser = null;
                 this.Clear();
             }
-            else
-            {
-                if (XboxLiveServicesSettings.Instance.DebugLogsOn)
-                {
-                    Debug.LogError(errorMessage);
-                }
-            }
         }
 
         private void OnEnable()
         {
-            this.SetupForAddingUser();
+            if (this.xboxLiveUser == null && SignInManager.Instance.GetCurrentNumberOfPlayers() > 0)

```

Comment:
Imagine this case

1) Scene is loaded and xboxLiveUser is set
2)  SetupForAddingUser() is called and adds a GetLeaderboardCompleted handler.
3) Scene disables the prefab
4) OnDisable() is called and removes GetLeaderboardCompleted handler
5) Scene enables the prefab
6) xboxLiveUser is not null and SetupForAddingUser() is not called again
7) User clicks next page but there is no GetLeaderboardCompleted handler set to receive the results

Reply:
I think a user would do this if they are showing different leaderboards on a single scene, they might just enable/disable the set of leaderboards.

---
In reply to: [182910556](https://github.com/Microsoft/xbox-live-unity-plugin/pull/223#discussion_r182910556) [](ancestors = 182910556)

Reply:
Should be fixed now.